components/email/src/Email.svelte

```diff
@@ -2063,13 +2061,41 @@
                 {/if}
               </div>
             </div>
-            {#if _this.show_received_timestamp}
-              <div class="message-date">
-                <span>
-                  {formatPreviewDate(new Date(_this.message.date * 1000))}
-                </span>
+            <section class="">
+              {#if _this.show_received_timestamp}
+                <div class="message-date">
+                  <span>
+                    {formatExpandedDate(new Date(_this.message.date * 1000))}
+                  </span>
+                </div>
+              {/if}
+              <div aria-label="Email Actions" role="toolbar">
+                {#if _this.show_reply}
+                  <div class="reply">
+                    <button
+                      title={"Reply"}
+                      aria-label={"Reply"}
+                      on:click|stopPropagation={(e) =>
+                        handleReplyClick(e, _this.message, "reply")}
+                    >
+                      <ReplyIcon aria-hidden="true" />
+                    </button>
+                  </div>
+                {/if}
+                {#if _this.show_reply_all && canReplyAll(_this.message)}
+                  <div class="reply-all">
+                    <button
+                      title={"Reply all"}
+                      aria-label={"Reply all"}
+                      on:click|stopPropagation={(e) =>
+                        handleReplyClick(e, _this.message, "reply_all")}
+                    >
+                      <ReplyAllIcon aria-hidden="true" />
+                    </button>
+                  </div>
+                {/if}
               </div>
-            {/if}
+            </section>
```

Comment:
Rather than adding duplicate code, this should be a component.

It doesn't follow the keyboard navigation for an element with `role="toolbar"`. You can see the design pattern [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/toolbar_role). I would recommend removing `role="toolbar"` for the time being, since adding the keyboard navigation will be time consuming. 

You will also need to add a focus style for these buttons as there currently is no indication of focus to the user.